azerothcore/azerothcore-wotlk
# fix(Core/AI): improve npc_escortAI

cp from tc

## Changes Proposed:
- NPC should not go into combat if it has REACT_PASSIVE
- NPC should not go into combat if the place is inaccessible
- NPC should not go into combat if AI cannot attack
- NPC should not go into combat if it is in EVADE mode
- NPC should not go into combat if the enemy is in EVADE mode
- Update DEFAULT_MAX_PLAYER_DISTANCE to 100 instead of 50 to allow players to be further away from the NPC (ref https://github.com/azerothcore/azerothcore-wotlk/issues/4860)

## Tests Performed:
- Build



## How to Test the Changes:

Edit by Shin: check a couple of creatures that use npc_escortAI

For example:

1. Get a creature that uses npc_escortAI
2. check the conditions above, creature should not aggro when they are true
3. Check that the NPC only despawns if you are >100 yards away

## Target Branch(es):








 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
